assignment7.txt

```diff
@@ -0,0 +1,19 @@
+Yael Kelmer. 
+
+1. ++*p increments the value of the datatype located at pointer p and then points at that location. *p++ points at the location in memory of p and then increments the value there.  
+
+2. The left to right order is guaranteed for operator precedence. 
+
+
+3. One advantage of pointers is that you can move along a string and point to different parts of it. 
+
+4.1 datatype: char array
+4.2 this is invalid
```

Comment:
This is actually valid! "xyz" is just an array of characters, so the "xyz"[1] is just accessing "y". Then you just subtract 'y' from it to get 0. This is because, if you can recall, a char is just a value mapped to a character!